lib/web3/filter.js

```diff
@@ -149,6 +150,13 @@ var Filter = function (options, methods, formatter, callback) {
         } else {
             self.filterId = id;
 
+            // check if there are get pending callbacks as a consequence
+            // of calling get() with filterId unassigned.
+            self.getPendingCallbacks.forEach(function (cb){
```

Comment:
Could you make the name change :)

could you check in your network tap, if he missing logs are also missing in the response from `eth_getFilterLogs`?

Maybe they are dropped by the log parser or something @debris any idea?